.github/workflows/gradle-fast-replay-tests.yml

```diff
@@ -122,13 +115,6 @@ jobs:
       - name: Setup Go Corset
         uses: ./.github/actions/setup-go-corset
 
-      - name: Run unit tests
```

Comment:
You can run unit tests and replay tests in parallel and still keep all the job definitions in the same yaml file. IMO the benefit of having one yaml file is that all the jobs that are run on every commit are defined in one single place. 

As of now there are 3 major job definitions in this file:

1. build
2. tests
3. go-corset-tests

Both tests and go-corset-tests depend on the build job and run in parallel after the build job completes.

As part of the tests and go-corset-tests we run unit tests and replay tests sequentially. One reason for doing this was to cut on unnecessary runs for replay tests in case the unit tests failed.

You can remove the tests (anb go-corset-tests) job(s) and introduce `unit_tests` and `replay_tests` jobs which can run in parallel after the build job is complete.

Reply:
Yeah, I understand what you are saying ... but having everything in one file can be crowded.  However, factoring out the common parts would make a big difference.  In fact, @FlorianHuc is looking into that separately as it happens.

Reply:
I will try to refactor it though, as I understand you're saying there's duplication which is not needed here.

Reply:
Also, are you familiar with this:

```
     - name: Setup upterm session
        if: ${{ inputs.tests-with-ssh }}
        uses: lhotari/action-upterm@v1
```

Do we ever actually use this?

Reply:
Ok, I have now put through a fairly heavy refactoring.  Hopefully that's better!